addons/mail/static/src/new/thread/link_preview/link_preview_list.js

```diff
@@ -0,0 +1,50 @@
+/** @odoo-module */
+
+import { useMessaging } from "../../messaging_hook";
+import { Component } from "@odoo/owl";
+import { LinkPreviewCard } from "./link_preview_card";
+import { LinkPreviewImage } from "./link_preview_image";
+import { LinkPreviewVideo } from "./link_preview_video";
+
+export class LinkPreviewList extends Component {
+    setup() {
+        this.messaging = useMessaging();
+    }
+
+    // TODO Could somehow directy use the
+    get linkPreviews() {
+        return Object.values(this.messaging.linkPreviews).filter(
+            (linkPreview) => linkPreview.message_id === this.props.message.id
+        );
+    }
+
+    get linkPreviewsImage() {
+        return this.linkPreviews.filter((linkPreview) =>
+            Boolean(linkPreview.image_mimetype || linkPreview.og_mimetype === "image/gif")
+        );
+    }
+
+    get linkPreviewsVideo() {
+        return this.linkPreviews.filter((linkPreview) =>
+            Boolean(
+                linkPreview.og_type.startsWith("video") &&
+                    !this.linkPreviewsImage.includes(linkPreview)
+            )
+        );
+    }
+
+    get linkPreviewsCard() {
+        return this.linkPreviews.filter((linkPreview) =>
+            Boolean(
+                !this.linkPreviewsVideo.includes(linkPreview) &&
+                    !this.linkPreviewsImage.includes(linkPreview)
+            )
+        );
+    }
+}
+
+Object.assign(LinkPreviewList, {
+    template: "mail.link_preview_list",
+    components: { LinkPreviewCard, LinkPreviewImage, LinkPreviewVideo },
+    props: ["message", "canBeDeleted"],
```

Comment:
I feel like `canBeDeleted` should be more appropriately optional.
That way, it's not as heavily coupled to `Message` component, so we could reuse this component without `Message`.